### PR TITLE
90059 disability benefits nb implement lighthouse document upload service

### PR DESCRIPTION
**Pull request by NB28VT**
## Summary

Completes the work started in https://github.com/department-of-veterans-affairs/vets-api/pull/17903 to follow other service's lead in migrating Form 526 ancillary document uploads to Lighthouse.

Finishes a Lighthouse Document Upload `ApiProvider` per the provider pattern implemented in the PR above, and creates a Form526-specific version of a service class for interacting with the Lighthouse API using the [API wrapper](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/lighthouse/benefits_documents/worker_service.rb) the mobile team built, which takes care of all the complex authentication used in this API. I say Form526-specific because there is already a [service class](https://github.com/department-of-veterans-affairs/vets-api/blob/90059-disability-benefits-nb-implement-lighthouse-document-upload-service/lib/lighthouse/benefits_documents/service.rb) for this API endpoint, but its functionality is too tightly tied to the claims status tool to be re-used for Form 526.

- *This work is behind a feature toggle (flipper):
It will be the behavior isn't used anywhere yet. It will be used in several different upload jobs
- *(Which team do you work for, does your team own the maintenance of this component?)*
Disability Benefits, and yes 

## Related issue(s)

https://github.com/department-of-veterans-affairs/va.gov-team/issues/90059

## Testing done

- [X] *New code is covered by unit tests*

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X]  No error nor warning in the console.
- [X]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable) not yet but will
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

**Comment by NB28VT** [comment]
Ah CI missed the codeowners change earlier need to address

**Review by aurora-a-k-a-lightning — APPROVED** [review]

**Review by aurora-a-k-a-lightning — DISMISSED** [review]

File: spec/support/vcr_cassettes/lighthouse/benefits_claims/documents/lighthouse_form_526_document_upload_200.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
+---
+http_interactions:
+- request:
+    method: post
+    uri: https://sandbox-api.va.gov/oauth2/benefits-documents/system/v1/token
```
Comment (RachalCassity) [inline spec/support/vcr_cassettes/lighthouse/benefits_claims/documents/lighthouse_form_526_document_upload_200.yml:5]: Can you please replace the domain with an ENV.  URI can be defined [here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/spec/support/vcr.rb#L9).

File: spec/support/vcr_cassettes/lighthouse/benefits_claims/documents/lighthouse_form_526_document_upload_400.yml [hunk 2]
```diff
@@ -0,0 +1,114 @@
+---
+http_interactions:
+- request:
+    method: post
+    uri: https://sandbox-api.va.gov/oauth2/benefits-documents/system/v1/token
```
Comment (RachalCassity) [inline spec/support/vcr_cassettes/lighthouse/benefits_claims/documents/lighthouse_form_526_document_upload_400.yml:5]: Same thing, mask URI

File: spec/support/vcr_cassettes/lighthouse/benefits_claims/documents/lighthouse_form_526_document_upload_200.yml (hunk 1 above)
Comment (NB28VT) [reply spec/support/vcr_cassettes/lighthouse/benefits_claims/documents/lighthouse_form_526_document_upload_200.yml]: Fixed in https://github.com/department-of-veterans-affairs/vets-api/pull/17991/commits/8ed3889a30bca07a4c54850163bfe9d8a8de6fdf, thanks for the heads up!

File: spec/support/vcr_cassettes/lighthouse/benefits_claims/documents/lighthouse_form_526_document_upload_400.yml (hunk 2 above)
Comment (NB28VT) [reply spec/support/vcr_cassettes/lighthouse/benefits_claims/documents/lighthouse_form_526_document_upload_400.yml]: Also fixed in https://github.com/department-of-veterans-affairs/vets-api/pull/17991/commits/8ed3889a30bca07a4c54850163bfe9d8a8de6fdf

**Review by RachalCassity — APPROVED** [review]

**Comment by NB28VT** [comment]
@RachalCassity thanks for the approval! I'm confused by the CI failures, is this safe to merge? It seems like it is?

**Comment by RachalCassity** [comment]
@NB28VT Yes

**Pull request merged**